### PR TITLE
chore(types): move to @types/yeoman-generator

### LINKED ISSUE #865: [Chore]: move types to @types/yeoman-generator
At the moment we use internal interfaces for our generator classes.

We should move to [`@types/yeoman-generator`](https://www.npmjs.com/package/@types/yeoman-generator)

### PR DESCRIPTION
**Pull request by misterdev**
**What kind of change does this PR introduce?**
Removes custom Yeoman types in favour of `@types/yeoman-generator`

**Summary**
Closes #865 

It compiles and seems to be working, I've marked with  `// TODO` where there's some work left to do

Any feedback is appreciated :)

### CODE REVIEW

File: packages/generators/add-generator.ts [hunk 1]
```diff
@@ -98,17 +98,19 @@ export default class AddGenerator extends Generator {
 				webpackOptions: {}
 			}
 		};
-		const { registerPrompt } = this.env.adapter.promptModule;
-		registerPrompt("autocomplete", autoComplete);
+		// TODO adapter doesn't exists
+		// const { registerPrompt } = this.env.adapter.promptModule;
+		// registerPrompt("autocomplete", autoComplete);
```
Comment (misterdev) [inline packages/generators/add-generator.ts]: Can anyone explain me what is this? :)

Comment (ematipico) [reply packages/generators/add-generator.ts]: @dhruvdutt might know something

Comment (misterdev) [reply packages/generators/add-generator.ts]: Some context, the type of env in [DefinitelyTyped ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/yeoman-generator/index.d.ts#L89) is:

```
    env: {
        error(...e: Error[]): void
    };
```

Comment (misterdev) [reply packages/generators/add-generator.ts]: The adapter is the part that actually handler the user interaction ([doc](https://yeoman.io/authoring/integrating-yeoman.html)).
`registerPrompt` is used to define a custom Inquirer prompt ([source](https://github.com/SBoudrias/Inquirer.js/blob/master/packages/inquirer/lib/inquirer.js#L46)).

I'm going to open a PR to include those in DefinitelyTyped, I just need to understand how to write tests :)

File: packages/generators/addon-generator.ts [hunk 2]
```diff
@@ -27,21 +25,20 @@ import { YeoGenerator } from "../generate-loader/types/Yeoman";
...
+const addonGenerator = (
+	prompts: Generator.Questions,
 	templateDir: string,
 	copyFiles: string[],
 	copyTemplateFiles: string[],
 	templateFn: Function
-): YeoGenerator {
-	return class AddOnGenerator extends Generator {
-		public props: InquirerScaffoldObject;
+): typeof Generator => class AddonGenerator extends Generator {
```
Comment (misterdev) [inline packages/generators/addon-generator.ts]: Not sure about `typeof Generator`, just `Generator` doesn't work

**Pull request merged**